### ISSUE TITLE
The global variable `--tw-content` should be in base instead of preflight

### ISSUE THREAD

**Issue by u3u** [tailwindlabs/tailwindcss#6486]
**What version of Tailwind CSS are you using?**

v3.0.2

**What build tool (or framework if it abstracts the build tool) are you using?**

tailwindcss-cli 3.0.2

**What version of Node.js are you using?**

v16.13.0

**What browser are you using?**

Chrome

**What operating system are you using?**

macOS

**Describe your issue**

If I set `corePlugins.preflight` to `false` all pseudo elements will be lost

https://github.com/tailwindlabs/tailwindcss/blob/0ddbb36c853059e19d779fbe66c65daccf3aef0e/src/css/preflight.css#L15-L18

**Comment by reinink** [comment]
Thanks for sharing this — you're right. This variable definition should be colocated with the other defaults (like transform, borders, etc).

**Comment by reinink** [comment]
Hey! So I've had a chance to dig into this a little further and spoke with the rest of the team, and the way this works is actually by design, and we've made similar choices for other features in Tailwind. 

For example, in Tailwind it's possible to use just the `border` utility to add a `1px` border to an element. This is possible because preflight sets a number of border defaults:

https://github.com/tailwindlabs/tailwindcss/blob/0ddbb36c853059e19d779fbe66c65daccf3aef0e/src/css/preflight.css#L10-L12

However, if you disable preflight, you'll need to add both `border` and `border-solid` to add the same 1px border:

```html
<div class="border border-solid">
  <!-- ... -->
</div>
```

In the same way, if you want to use the `before` and `after` modifiers in a project with preflight disabled, you'll need to manually set the content to an empty string.

You can do this either with an arbitrary value (as noted above):

```html
<div class="after:content-[''] after:display ...">
  <!-- ... -->
</div>
```

Or you can do this by adding the variable declaration to your own CSS:

```css
::before, 
::after { 
  --tw-content: ''; 
} 
```

I put together a little play demo to illustrate this: https://play.tailwindcss.com/nThK37dCQy

Basically, if you include Tailwind's default styles, things like borders and the content property become a little easier, as we handle this extra stuff for you.

Maybe there's a chance that we could make this work without preflight in the future — but for now this is what we recommend. And even if we did change this in the future, your code will still work, so no worries there.

I've also added a note about this to [the docs](https://tailwindcss.com/docs/hover-focus-and-other-states#before-and-after) to make sure this is clear for anyone else who runs into this issue.

I hope that helps! 💪

**Comment by u3u** [comment]
Understood, but this may cause problems for people upgrading from v2, and may need to be clarified in the upgrade guide
